docs/reference/api/platform/Error.md

```diff
@@ -1,14 +1,123 @@
+**Although we support Posix error codes for compatibility reasons, its highly encouraged that all future Mbed-OS focussed implementations use Mbed-OS error definitions so that errors reported works seamlessly with error reporting and handling implementation in Mbed-OS.**
+
+### Error Context Capture
+Error handling system in Mbed OS automatically captures the thread context for each error reported into the system. The context captured includes following information:
+
+- Error Status Code – Status code
+- Error Address – Captures return address of set error calls.
+- Error Value – A context specific value set by the caller in MBED_ERROR()/MBED_WARNING() calls
+- Thread ID – ID of the current thread - This information is captured only if the system is built with RTOS support.
+- Thread Entry Address – Entry function for the current thread - This information is captured only if the system is built with RTOS support.
+- Thread Stack Size – Size of stack - This information is captured only if the system is built with RTOS support.
+- Thread Stack Mem - Top of stack - This information is captured only if the system is built with RTOS support.
+- Thread Current SP – SP value (PSP or MSP based on what’s active) - This information is captured only if the system is built with RTOS support.
+- Filename and Line number - By default capturing of filename, line number is disabled. If you need this information to be captured it needs to be enabled using **MBED_CONF_ERROR_FILENAME_CAPTURE_ENABLED** macro.
+
+This captured context will be emitted through STDOUT in the case of fatal errors and in the case of warnings it will be recorded by the system and can be retrieved later for system diagnostics, external reporting and for debugging purposes.
+
+### Error Reporting
+Mbed OS provides mainly three ways to report an error using the error handling implementation - 
+1. Using **error()** function to report a fatal error condition - This is the legacy function used for reporting a fatal error. It has been modified to capture the context information. But, it will not capture the address of the caller, module information and also doesn't support reporting specific error codes. Any implementation using this API to report an error will use the error status **MBED_ERROR_UNKNOWN**. Its highly encouraged that all future Mbed-OS focussed implementations use MBED_ERROR()/MBED_ERROR1() or MBED_WARNING()/MBED_WARNING() macros for reporting errors.**
```

Comment:
1. Whats does the `**` reference?
2. Should MBED_WARNING()/MBED_WARNING() be something else after the slash?

Reply:
@kegilbert  - I'm using ** is just used to making them bold letters. I'll fix the MBED_WARNING line.

Reply:
Sorry realized how big the quoted section is. There's a trailing ** at the very end, thought it was for a post note at first but looks like just an extra bold point ending "bracket".